controllers/k8s/pod_log_reader.go

```diff
@@ -0,0 +1,40 @@
+package k8s
+
+import (
+	"bytes"
+	"github.com/go-errors/errors"
+	"github.com/redhatinsights/xjoin-go-lib/pkg/utils"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type LogReader interface {
+	GetLogs(name string, namespace string) (string, error)
+}
+
+type PodLogReader struct {
+	ClientSet kubernetes.Interface
+}
+
+func (p PodLogReader) GetLogs(name string, namespace string) (string, error) {
```

Comment:
Nit: I'd switch the order of the arguments, to have first `namespace` and `name` second.